stdlib/universe/aggregate_window.gen.go

```diff
@@ -21,13 +21,30 @@ func (a *aggregateWindowSumInt) Aggregate(ts *array.Int, vs array.Array, start,
 	b := array.NewIntBuilder(mem)
 	b.Resize(stop.Len())
 
+	// Check once if we should look for nulls in the input.
+	hasNulls := vs.NullN() > 0
+
 	values := vs.(*array.Int)
 	aggregateWindows(ts, start, stop, func(i, j int) {
-		var sum int64
+		var (
+			sum    int64
+			isNull = true
+		)
 		for ; i < j; i++ {
+			// If there are nulls, check if this is null.
+			if hasNulls && values.IsNull(i) {
```

Comment:
```suggestion
			if values.IsNull(i) {
```
Checking `hasNulls` is just a redundant extra branch on each index (unless the optimizer does some magic?). I suppose `hasNulls` could be checked outside the loop though and branch on that.

```go
		if hasNulls {
			for ; i < j; i++ {
				// If there are nulls, check if this is null.
				if values.IsNull(i) {
					continue
				}
				sum += values.Value(i)
				isNull = false
			}
		} else {
			for ; i < j; i++ {
				sum += values.Value(i)
				isNull = false
			}
		}

```

Reply:
Changed in both sum and mean.